gap/base/recognition.gd

```diff
@@ -563,9 +563,12 @@ DeclareSynonym("RecognizeGroup", RecogniseGroup);
 ##     components of which are copied into the new recognition info record
 ##     which is created for the group <A>H</A>. Especially the component
 ##     <C>hints</C> can contain a list of additional find homomorphism
-##     methods (described by records as in Section <Ref Sect="whataremethods"/>)
-##     which is prepended to the method database in <A>methoddb</A> before
-##     the recognition starts. This feature is intended to give hints
+##     methods (described by records as in Section <Ref Sect="whataremethods"/>).
+#      The methods in <C>hints</C> and in <A>methoddb</A> are merged and sorted
+#      into rank-descending order. The result is passed to
+#      <Ref Func="CallMethods"/>. Thus you can force the methods in
```

Comment:
This sound strange to me. You "can" force this? Isn't it always true that `hints` methods are called first? At the same time, this is only one of the ways you can use `hints` for, you can also use it to add methods which are not applicable in general, but nevertheless might not have the highest rank. So why call out that use case especially? But if you really think it's useful, I'd still rephrase it...

Reply:
I'll drop the "can force" sentence. It's not like it adds anything valuable.

> Isn't it always true that hints methods are called first? 

This is why I actually did this commit. At the moment, the hint methods are _not_ always called first, despite what the documentation says. This is the responsible [line of code](https://github.com/gap-packages/recog/blob/678f1317332e152fe5d718231a0ecdb78e4872ab/gap/base/recognition.gi#L420). But I ran git blame a little and I have the feeling that the original "sorting the hints and methoddb into rank-descending order" was done on accident in commit 30717370 with title "Add example of successful recognition". I guess that commit should only have touched  `doc/afterrecog.xml`.

Reply:
I've moved the respective commit into PR #187